### DISCUSSION TITLE
monailabel returns PYTHONPATH

### DISCUSSION THREAD

**Discussion by marcosdornellas** [Project-MONAI/MONAILabel#1130] [Q&A]
Hi folks,

I don't understand why terminal responds with "Using PYTHONPATH=/Library/Frameworks/Python.framework/Versions:" to all commands to installed app
Apologies, because I feel this is going to be a really stupid question.

I have installed a framework call Monai Label via Terminal ([Installation — MONAI Label 0.4.2 Documentation](https://docs.monai.io/projects/label/en/stable/installation.html)).

Working my way through the prerequisites I have successfully updated Python to 3.8.9, as well as PyTorch. I have successfully installed MonaiLabel via pip install.

But now whenever I type in a command to MonaiLabel (like --version, or --help) terminal responds with Using PYTHONPATH=/Library/Frameworks/Python.framework/Versions:.

Thinking this meant I had to update the PYTHONPATH variable to the given path I updated the bash_profile via atom ~/.bash_profile. This file saved successfully, but that didn't seem to do anything.

Thanks in advance,
marcos

**Comment by tangy5** [comment]
Thanks for the question, monailabel will automatically set PYTHONPATH to the ENV , where apps or lib located

See here in the init setting for monailabel
https://github.com/Project-MONAI/MONAILabel/blob/5a076c8a08ce18244681f1b0a748db4f8807da4a/monailabel/main.py#L283

I don't think you will beed to modify the bash file, but I guess you might want to look at that for checking the python version. 
Thanks

**Reply by marcosdornellas** [reply]
Hi Tangy5!

Thanks for the kind answer. 
When I type in the command line:
% monailabel --help

I just get:
Using PYTHONPATH=/Library/Frameworks/Python.framework/Versions:

I have just followed the installation in https://docs.monai.io/projects/label/en/latest/installation.html

user@WhiteLynx ~ % python3 -c "import monai; monai.config.print_config()"
MONAI version: 1.0.1
Numpy version: 1.23.0
Pytorch version: 1.13.0
MONAI flags: HAS_EXT = False, USE_COMPILED = False, USE_META_DICT = False
MONAI rev id: 8271a193229fe4437026185e218d5b06f7c8ce69
MONAI __file__: /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/monai/__init__.py

Optional dependencies:
Pytorch Ignite version: 0.4.10
Nibabel version: 4.0.1
scikit-image version: 0.19.3
Pillow version: 9.1.1
Tensorboard version: 2.9.1
gdown version: 4.5.3
TorchVision version: 0.14.0
tqdm version: 4.64.0
lmdb version: 1.3.0
psutil version: 5.9.1
pandas version: 1.4.3
einops version: 0.5.0
transformers version: 4.24.0
mlflow version: 1.30.0
pynrrd version: 0.4.3

**Comment by tangy5** [comment]
emm, you should see something like below after the PYTHONPATH:

```
Using PYTHONPATH=/home/yucheng:

usage: monailabel [-h] [-v] {start_server,apps,datasets,plugins} ...

positional arguments:
  {start_server,apps,datasets,plugins}
                        sub-command help
    start_server        Start Application Server
    apps                list or download sample apps
    datasets            list or download sample datasets
    plugins             list or download viewer plugins

optional arguments:
  -h, --help            show this help message and exit
  -v, --version         print version
```

Which operating system are you using? If you followed the install page, did you install monailabel by "pip"? or from Github?

**Reply by marcosdornellas** [reply]
Good morning,

user@WhiteLynx monailabel % sw_vers        
ProductName:		macOS
ProductVersion:		13.0.1
BuildVersion:		22A400

I've tried both... Same results.

**Reply by marcosdornellas** [reply]
Hi,

I figured out now... It worked with my Ubuntu server. Many thanks.

**Reply by MattTheCuber** [reply]
What was your solution?

**Comment by dhudsmith** [comment]
I had the same issue on Ubuntu 22.04, WSL 2. In my case, the issue was that monailabel expected the python binary at /usr/bin/python and I only had /usr/bin/python3. This can be fixed with a symlink or equivalently by `sudo apt install python-is-python3`.

**Reply by bunengzaicai** [reply]
Thank you very much. Your solution is useful and important for me.

**Comment by evaherbst** [comment]
I had the same issue. It was returning the python path (which is expected behaviour) but then just stalling indefinitely (I cut it after about 10 minutes). For me, the issue seemed to be due to the following:

I am not sure why (since I was running MonaiLabel locally) but it seems that the cause of my installation issues was due to the university proxy (which I was told to use to access the internet).
I instead purchased a WIFI dongle and used WIFI (to avoid the proxy). I installed monailabel in a miniconda venv on Windows and everything worked well.

**Reply by diazandr3s** [reply]
Thanks for the update, @evaherbst
